src/sandbox/components/screens/call-functions/components/available-functions.tsx

```diff
@@ -177,8 +183,26 @@ const FunctionSingleView = () => {
   const { handleTrackGoal } = useFathomGoal();
   const network = useNetworkConfig();
 
-  const { handleSubmit, handleChange, setValues, values } = useFormik({
+  console.log(11111111, 'FunctionSingleView');
+
+  const { handleSubmit, handleChange, setValues, values, errors } = useFormik({
     initialValues,
+    validateOnChange: false,
+    validateOnBlur: false,
+    validate: values => {
+      const errors = {};
+      Object.keys(values).forEach(key => {
+        const type = fn.args.find(({ name }) => name === key).type;
+        const optionalType = type?.optional;
+        if (type === 'principal' || (optionalType === 'principal' && !!values[key])) {
+          const validPrincipal = validateStacksAddress(values[key]);
+          if (!validPrincipal) {
+            errors[key] = 'Invalid Stacks address.';
```

Comment:
A nitpick; perhaps remove the `.` from the error message (haha). If that is the format followed today for all messages, please keep it as-is.

Reply:
Yep we have the same exact message in other places🙂, we should revisit this at some point.